scheduler/scheduler-server/src/main/java/org/ow2/proactive/utils/JettyStarter.java

```diff
@@ -71,73 +84,171 @@ public static void main(String[] args) {
+            int httpsPort = Integer.parseInt(properties.getProperty(WebProperties.WEB_HTTPS_PORT, "443"));
+
+            boolean httpsEnabled = isHttpsEnabled(properties);
+            boolean redirectHttpToHttps = isHttpToHttpsRedirectionEnabled(properties);
+
+            int restPort = httpPort;
+
+            String httpProtocol;
+            String[] defaultVirtualHost;
+            String[] httpVirtualHost = new String[] { "@" + HTTP_CONNECTOR_NAME };
+
+            if (httpsEnabled) {
+                httpProtocol = "https";
+                defaultVirtualHost = new String[] { "@" + HTTPS_CONNECTOR_NAME };
+                restPort = httpsPort;
+            } else {
+                defaultVirtualHost = httpVirtualHost;
+                httpProtocol = "http";
+            }
+
+            Server server =
+                    createHttpServer(
+                            properties, httpPort, httpsPort, httpsEnabled, redirectHttpToHttps);
+
             server.setStopAtShutdown(true);
 
             HandlerList handlerList = new HandlerList();
-            addWarsToHandlerList(handlerList);
+
+            if (redirectHttpToHttps) {
+                ContextHandler redirectHandler = new ContextHandler();
+                redirectHandler.setContextPath("/");
+                redirectHandler.setHandler(new SecuredRedirectHandler());
+                redirectHandler.setVirtualHosts(httpVirtualHost);
+                handlerList.addHandler(redirectHandler);
+            }
+
+            addWarsToHandlerList(handlerList, defaultVirtualHost);
             server.setHandler(handlerList);
 
-            String schedulerHost = ProActiveInet.getInstance().getHostname();
-            startServer(server, schedulerHost, restPort, httpProtocol);
+            startServer(server, ProActiveInet.getInstance().getHostname(), restPort, httpProtocol);
         }
     }
 
-    private static Server createHttpServer(Properties properties, int restPort, boolean httpsEnabled) {
-        int maxThreads = Integer.parseInt(properties.getProperty("web.max_threads", "100"));
-        QueuedThreadPool threadPool = new QueuedThreadPool(maxThreads);
+    protected int getJettyHttpPort(Properties properties) {
+        int result = 8080;
+
+        String property = properties.getProperty(WebProperties.WEB_HTTP_PORT);
+
+        if (property == null) {
+            property = properties.getProperty(WebProperties.WEB_PORT);
+        }
+
+        if (property != null) {
+            result = Integer.parseInt(property);
+        }
+
+        return result;
+    }
 
+    private boolean isHttpToHttpsRedirectionEnabled(Properties properties) {
+        return properties.getProperty(
+                WebProperties.WEB_REDIRECT_HTTP_TO_HTTPS, "true").equalsIgnoreCase("true");
+    }
+
+    private boolean isHttpsEnabled(Properties properties) {
+        return properties.getProperty(WebProperties.WEB_HTTPS, "false").equalsIgnoreCase("true");
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "true" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://master.ci.activeeon.com:9000/coding_rules#rule_key=squid%3AS1132)